### ISSUE TITLE
Question about reproducing CIFAR-10 results

### ISSUE THREAD

**Issue by wsgharvey** [SHI-Labs/Compact-Transformers#45]
Hi, thank you for this very clean open-source implementation!

I've been testing out some modifications and noticed, even without the modifications, I'm not quite achieving the same accuracies as you report on CIFAR-10. I wondered if you had any suggestions about things that I might have missed.

Specifically, I tried to reproduce your results using one of your configs with the command `python train.py -c configs/pretrained/cct_7-3x1_cifar10_300epochs.yml --model cct_7_3x1_32 datasets/CIFAR-10-images/ --log-wandb`. I copied the dataset from [this github repo](https://github.com/YoongiKim/CIFAR-10-images). I couldn't find details on whether you use a train/validation split so then trained on all 50000 training images (i.e. with no validation set) and tested on all 10000 test images. I used your `validate` function for computing the test accuracy (by renaming the test image folder so that it is loaded as a validation set). After the full 300 epochs, I obtained 93.21% test accuracy, rather than the 96.53% that I think you report for this config in the README. Please let me know if there's anything I should do differently to obtain these results - perhaps using a different train/test split, computing the test loss in a different way, turning on EMA averaging, or if there's anything else that I might have missed.

I also tried computing test statistics in the same way after loading your pretrained `cct_7_3x1_32` checkpoint (instead of training it) and got a lower test accuracy of 91.67%. So this makes me think that the issue is likely related to testing rather than training.

Thanks!

**Comment by alihassanijr** [comment]
Hi,
Thank you for your interest.
We used the CIFAR-10/100 classes from torchvision (not supported by the version of `timm` that was available when we last updated this repository). We were planning on updating this repo to support those datasets, but apparently now the latest version of timm supports torchvision datasets.

We are going to update this repo soon, but until then we highly recommend using the torchvision version. All you have to do is change these lines, for instance add an if statement for those datasets and import them:
https://github.com/SHI-Labs/Compact-Transformers/blob/e6869b4326e252bf9d6c1e5439ab824f2f4d7f2a/train.py#L476-L481

There is an example in another repository of ours that you can refer to:
https://github.com/SHI-Labs/Convolutional-MLPs/blob/89008c686e48803c012038f21f97e56276aa84ad/classification.py#L481-L506

Please let us know if this resolves your issue.
I'll be closing this issue for now, but feel free to open it back up.

**Comment by wsgharvey** [comment]
Thanks for the quick reply! I'll give that a try

**Comment by wsgharvey** [comment]
Still very confused about what is different between these two ways of loading the dataset, but replacing those lines with:
```
from torchvision.datasets import CIFAR10
dataset_train = CIFAR10(root='.', train=True, download=True)
dataset_eval = CIFAR10(root='.', train=False, download=True)
```
 has got me to 96.21% accuracy, so close to your reported value. Thanks for your help!

**Comment by stevenwalton** [comment]
The first is the training dataset and the second is the validation set. You can read more on the format [here](https://pytorch.org/vision/stable/datasets.html) on Torchvision's official documentation.

**Comment by wsgharvey** [comment]
To clarify, my confusion is about the difference between the images loaded using `torchvision.datasets.CIFAR10` and using `create_dataset` with a directory of images (as I had been doing before). They both seem to be normalised to roughly zero mean and unit variance, and look similar when plotted with `--save-images`. Anyway, it's working with the torchvision dataset so maybe there's no need to dig too deeply into what went wrong before. Cheers
